common/primitive-types/src/primitives.rs

```diff
@@ -639,4 +661,74 @@ mod tests {
         assert_eq!(expected_le, le_bytes);
         assert_eq!(U256::from_le_bytes(expected_le), num);
     }
+
+    #[test]
+    fn test_address_to_checksum_all_caps() -> anyhow::Result<()> {
+        let addr_1 = Address::from_str("52908400098527886e0f7030069857d2e4169ee7")?;
+        let value_1 = addr_1.to_checksum();
+        let addr_2 = Address::from_str("8617e340b3d01fa5f11f306f4090fd50e238070d")?;
+        let value_2 = addr_2.to_checksum();
+
+        assert_eq!(
+            value_1, "0x52908400098527886E0F7030069857D2E4169EE7",
+            "checksumed address does not match"
+        );
+        assert_eq!(
+            value_2, "0x8617E340B3D01FA5F11F306F4090FD50E238070D",
+            "checksumed address does not match"
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_address_to_checksum_all_lower() -> anyhow::Result<()> {
+        let addr_1 = Address::from_str("de709f2102306220921060314715629080e2fb77")?;
+        let value_1 = addr_1.to_checksum();
+        let addr_2 = Address::from_str("27b1fdb04752bbc536007a920d24acb045561c26")?;
+        let value_2 = addr_2.to_checksum();
+
+        assert_eq!(
+            value_1, "0xde709f2102306220921060314715629080e2fb77",
+            "checksumed address does not match"
+        );
+        assert_eq!(
+            value_2, "0x27b1fdb04752bbc536007a920d24acb045561c26",
+            "checksumed address does not match"
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_address_to_checksum_all_normal() -> anyhow::Result<()> {
```

Comment:
I'd prefer the test names to actually describe what is being tested in more detail, so that a test failure is self-descriptive. The `test_` is not necessary in rust at all.

Reply:
This is just a copy paste of pre-existing tests.